### ISSUE TITLE
Feature Request: Add ELB logging configuration when Load Balanced Web Service Type

### ISSUE THREAD

**Issue by nao-kagayaki** [aws/copilot-cli#1530]
Hi, I  assessed the copilot to use copilot for my future project.
In production use,  log of the Elastic Load Balancing is crucial for trobleshooting.
Could you tell me whether if you plan to add ELB logging configuration features? If not I would like to request this feature.

**Comment by efekarakus** [comment]
Heya @nyamada43 !

This would be a new feature request for us!
Do you have any practices in particular that you follow for elb logging configuration? For example around log retention.

This might be helpful for us to look for implementation: https://github.com/awslabs/aws-cloudformation-templates/blob/master/aws/services/ElasticLoadBalancing/ELB_Access_Logs_And_Connection_Draining.yaml#L403-L426

**Comment by nao-kagayaki** [comment]
Thank you for your comment @efekarakus

I think in many cases I want to set LifeCycleConfiguration to match log preservation policies of businesses.
Although this is not must for production use, it would be nice if it's supported in the future.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig.html

To elaborate on my image, if the user selects the Load Blanced web service, its configuration file is as follows.
First of all, it would be helpful if we could support the ability to store ELB logs in s3.

```
name: frontend
type: Load Balanced Web Service

image:
  build: ./Dockerfile
  port: 80

http:
  path: '/'
  healthcheck: "/"
  stickiness: true

  accessLogsS3: true
```

**Comment by efekarakus** [comment]
Hi @nyamada43, @ceturc, @artishan, @webjunkie!

Out of curiosity, for environments where you enable ELB logging would you also enable container insights for the cluster and VPC flow logs?
What would you think of an approach where running `copilot env init --prod` by default enabled all these observability settings instead of specifying them individually?

**Comment by ceturc** [comment]
Hi @efekarakus , Good question. I do agree all the features you listed are reasonable defaults for a "--prod" setting for observability in an opinionated tool like copilot; your idea makes sense to me.

I personally would not use "--prod" unless I clearly understood what it did, per #2684 and #2158, so I could opt-out of any behavior I don't need in the opinionated defaults.   I don't have a requirement for container insights and VPC flow logs in my use case, so I would opt-out of those.

I do like the copilot strategy of reasonable opinionated defaults that can be overridden.   Thanks for sharing the idea - hope it moves forward.

> Out of curiosity, for environments where you enable ELB logging would you also enable container insights for the cluster and VPC flow logs? What would you think of an approach where running `copilot env init --prod` by default enabled all these observability settings instead of specifying them individually?

**Comment by efekarakus** [comment]
Thanks for the feedback @ceturc, it makes total sense!

Would you want to re-use the same `--prod` configuration when initializing environments? Or would you prefer to individually choose what to enable every time you create a new environment?

I'm trying to understand if we should optimize for a use case of "I'd like to define a prod config where container insights is enabled, and then reuse it every time I create an environment" vs. "I'd like to choose container insights for env1 but container insights + access logs for env2"

**Comment by ceturc** [comment]
@efekarakus Interesting question.  It's less cognitive load for me if `--prod` always works the same way (AWS best practices), and I can override the behavior for each environment.

**Comment by PeterBengtson** [comment]
Logs are not just required in prod environments. We need them in every environment, so an informal automatic dependency on something arbitrarily called "prod" would not be ideal. It's much better to declare these things each time, not relying in parsing environment names to enable it.

So, I'm _against_ a --prod switch.

**Comment by efekarakus** [comment]
👍 The issue was created before we released the new [Environment Manifest](https://aws.github.io/copilot-cli/docs/manifest/environment/) feature in [v1.20](https://github.com/aws/copilot-cli/releases/tag/v1.20.0).

Indeed the plan now is to support it as just a field in the infrastructure as code file  without having some dependency to a flag.
```yaml
name: qa
type: Environment
http:
  public:
    # access_logs: true # for basic configuration that will create a new bucket for access logs.
    access_logs:        # for advanced configuration to use existing infrastructure.
       interval: 30m
       bucket_name: my-bucket
       bucket_prefix: access-logs
```

**Comment by PeterBengtson** [comment]
That's ideal. :)

**Comment by dannyrandall** [comment]
This feature is now released in v1.21.0! https://github.com/aws/copilot-cli/releases/tag/v1.21.0

**Comment by rickychew77** [comment]
Hello, revisiting this issue that if internal load balancer will be covered with access_logs like how public load balancer does?
